dotCMS/src/integration-test/java/com/dotcms/storage/ContentletMetadataAPITest.java

```diff
@@ -0,0 +1,367 @@
+    private void validateBasic(final Map<String, Object> metaData){
+        assertTrue(metaData.containsKey("contentType"));
+        assertTrue(metaData.containsKey("modDate"));
+        assertTrue(metaData.containsKey("sha256"));
+        assertTrue(metaData.containsKey("path"));
+        assertTrue(metaData.containsKey("title"));
+    }
+
+    /**
+     * validate basic layout expected in the basic md for File-Asset
+     * But nothing else if there are additional values we fail!!
+     * @param metaData
+     */
+    private void validateBasicStrict(final Map<String, Object> metaData){
+        validateBasic(metaData);
+        assertEquals("we're expecting exactly 5 entries.",5,metaData.size());
+    }
+
+    //SVG  do not have dimensions, but that's a known issue we're willing to forgive.
+    private static Set<String> imageExt = ImmutableSet.of("jpg", "png", "gif");
+
+    /**
+     * Custom is image method. this custom version skips svg files.
+     * @param fileName
+     * @return
+     */
+    private boolean isSupportedImage(final String fileName) {
+        final String assetNameExt = UtilMethods.getFileExtension(fileName).toLowerCase();
+        return imageExt.contains(assetNameExt);
+    }
+
+    /**
+     * Given scenario: We have an instance of a content-type that has different fields of type bin
+     * this time we test that te first field get the generated full-md generated while the rest only get the basic version.
+     * Expected Results:
+     * @throws IOException
+     */
+    @Test
+    @UseDataProvider("getStorageType")
+    public void Test_Generate_Metadata_From_ContentType_With_Multiple_Binary_Fields(final StorageType storageType) throws IOException {
+        final long langId = APILocator.getLanguageAPI().getDefaultLanguage().getId();
+        final String stringProperty = Config.getStringProperty(DEFAULT_STORAGE_TYPE);
+        try {
+            Config.setProperty(DEFAULT_STORAGE_TYPE, storageType.name());
+
+            //Multiple binary fields
+            final Contentlet multipleBinariesContent = TestDataUtils
+                    .getMultipleBinariesContent(true, langId, null);
+
+            //Multiple binary fields
+            final ContentletMetadata multiBinaryMetadata = contentletMetadataAPI
+                    .generateContentletMetadata(multipleBinariesContent);
+            assertNotNull(multiBinaryMetadata);
+
+            final Map<String, Map<String, Object>> fullMetadataMap = multiBinaryMetadata
+                    .getFullMetadataMap();
+            assertNotNull(fullMetadataMap);
+
+            final Map<String, Map<String, Object>> basicMetadataMap = multiBinaryMetadata
+                    .getBasicMetadataMap();
+            assertNotNull(basicMetadataMap);
+
+            //the filed is set as the first one according to the sortOrder prop. This is the only that has to have full metadata
+            final Map<String, Object> fileAsset2FullMeta = fullMetadataMap.get("fileAsset2");
```

Comment:
Codacy found an issue: [The String literal "fileAsset2" appears 4 times in this file; the first occurrence is on line 180](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6220272)